### DISCUSSION TITLE
fits extension syntax

### DISCUSSION THREAD

**Discussion by mgregg128** [iraf-community/iraf#321] [Q&A]
The latest version of iraf appears unable to recognize the fits extension syntax of HST data, such as image.fits[sci,1] or image.fits[sci,2].   Is there a way to implement this?  The workaround is to specify the absolute extension number of course (sci,2= extension 4 in most instances), but it would be handy to have the sci,n syntax work, if only to be able to use older scripts.

I'm running iraf 2.17 on a 2017 MBP with OS Ventura.

Thank you for the help,
Michael

**Comment by olebole** [comment]
The problem is that 2.17 accidently was configured to use the old image-open procedure. The best way is to update to [version 2.17.1](https://github.com/iraf-community/iraf/releases/tag/v2.17.1), which has this fixed (https://github.com/iraf-community/iraf/pull/236). Alternatively, you can fix this by adding these lines to `unix/hlib/zzsetenv.def`:
```sh
set	use_vo		= yes 
set	use_new_imt	= yes 
```
This will switch to the extended image-open procedure that allows to specify the MEF extension.

**Reply by mgregg128** [reply]
Thank you for the advice; I will try it.
I almost immediately also found that using all-caps works too, as in image.fits[SCI,1], which is recognized by iraf 2.17

**Reply by mgregg128** [reply]
I tried adding those lines to zzsetenv.def.  This appears to have no effect.
If I issue the command:
>imhead image.fits[sci,1] 
iraf returns the message:
image.fits: FXF: must specify which FITS extension (image.fits)

But if I capitalize 'SCI', it works: 
image.fits[1]{4096,2051][real]:

So I will upgrade to 2.17.1 next.  
Thank you

**Reply by olebole** [reply]
If it doesn't work, the best here would as well be a small complete example that shows the problem.
